packages/apollo-tools/src/buildServiceDefinition.ts

```diff
@@ -65,6 +70,14 @@ export function buildServiceDefinition(
         } else {
           typeExtensionsMap[typeName] = [definition];
         }
+      } else if (definition.kind === Kind.DIRECTIVE_DEFINITION) {
+        const typeName = definition.name.value;
```

Comment:
This isn't really a `typeName` but a `directiveName`. If we're using a map, we should probably also enforce a directive to only be defined once and report an error otherwise. That would actually make sense I think, but [`buildASTSchema`](https://github.com/graphql/graphql-js/blob/64b194c6c9b9aaa1c139f1b7c3692a6ef851928e/src/utilities/buildASTSchema.js#L146-L148) doesn't do that either (it just keeps an array of directives) and that results in 'last wins'.

Reply:
```suggestion
        const directive = definition.name.value;
```